apps/emqx_connector/src/mqtt/emqx_connector_mqtt_msg.erl

```diff
@@ -151,8 +153,6 @@ estimate_size(#{topic := Topic, payload := Payload}) ->
 estimate_size(Term) ->
     erlang:external_size(Term).
 
-set_headers(undefined, Msg) ->
```

Comment:
If I understand it correctly, this clause should be kept.

Reply:
`dialyzer` says that
```
Line 156 Column 1: The pattern <'undefined', Msg> can never match the type <#{'properties':=map(), _=>_},#message{id::binary(),from::atom() | binary(),flags::#{atom()=>boolean()},headers::#{'allow_publish'=>boolean(), 'peerhost'=>{_,_,_,_} | {_,_,_,_,_,_,_,_}, 'properties'=>map(), 'proto_ver'=>binary() | non_neg_integer(), 'protocol'=>atom(), 'username'=>'undefined' | binary(), atom()=>_},topic::binary(),payload::binary() | maybe_improper_list(binary() | maybe_improper_list(any(),binary() | []) | byte(),binary() | []),timestamp::integer()}>
```

Reply:
With this clause, it's no longer possible to be `undefined`. This comment is resolved.
```
to_broker_msg(Msg, Vars, undefined) ->
    to_broker_msg(Msg, Vars, #{});
```